spring-integration-core/src/main/java/org/springframework/integration/gateway/MessagingGatewaySupport.java

```diff
@@ -533,6 +546,135 @@ private Object doSendAndReceive(Object object, boolean shouldConvert) {
 		return reply;
 	}
 
+	protected Mono<Message<?>> sendAndReceiveMessageReactive(Object object) {
+		initializeIfNecessary();
+		Assert.notNull(object, "request must not be null");
+		MessageChannel requestChannel = getRequestChannel();
+		if (requestChannel == null) {
+			throw new MessagingException("No request channel available. Cannot send request message.");
+		}
+
+		registerReplyMessageCorrelatorIfNecessary();
+
+		return doSendAndReceiveMessageReactive(requestChannel, object, false);
+	}
+
+	@SuppressWarnings("unchecked")
+	private Mono<Message<?>> doSendAndReceiveMessageReactive(MessageChannel requestChannel, Object object,
+			boolean error) {
+
+		AtomicReference<Message<?>> requestMessageReference = new AtomicReference<>();
```

Comment:
No need for this.
Move lines 623 `doOnSubscribe` and 628 `onErrorResume` inside the `defer`, just after line 621 `map`